platform/darwin/src/MGLImageSource.h

```diff
@@ -0,0 +1,136 @@
+#import "MGLSource.h"
+
+#import "MGLFoundation.h"
+#import "MGLTypes.h"
+#import "MGLGeometry.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+MGL_EXPORT
+/**
+ `MGLImageSource` is a map content source that supplies a georeferenced raster
+ image to be shown on the map. The geographic location of the raster image content, 
+ supplied with `MGLCoordinateQuad`, can be non-axis aligned.
+ `MGLImageSource` supports raster content from `NSURL` or an `MGLImage`.
+ Supported image formats are : PNG, TIFF, JPEG, GIF (single frame rendering),
+ and BMP.
+ An image source is added to an `MGLStyle` object along with one or more
+ `MGLRasterStyleLayer` objects. Use a raster style layer to control the
+ appearance of content supplied by the image source.
+ 
+ Each
+ <a href="https://www.mapbox.com/mapbox-gl-style-spec/#sources-image"><code>image</code></a>
+ source defined by the style JSON file is represented at runtime by an
+ `MGLImageSource` object that you can use to initialize new style layers. You
+ can also add and remove sources dynamically using methods such as
+ `-[MGLStyle addSource:]` and `-[MGLStyle sourceWithIdentifier:]`.
+
+ ### Example
+
+ ```swift
+ let coordinates: MGLCoordinateQuad = MGLCoordinateQuad(
+   topLeft: CLLocationCoordinate2D(latitude: 46.437, longitude: -80.425),
+   topRight: CLLocationCoordinate2D(latitude: 46.437, longitude: -71.516),
+   bottomRight: CLLocationCoordinate2D(latitude: 37.936, longitude: -71.516),
+   bottomLeft: CLLocationCoordinate2D(latitude: 37.936, longitude: -80.425))
+ let source = MGLImageSource(identifier: "images", coordinates: coordinates, imageURL: URL(string: "https://www.mapbox.com/mapbox-gl-js/assets/radar.gif")!)
+ mapView.style?.addSource(source)
+ ```
+ */
+MGL_EXPORT
+@interface MGLImageSource : MGLSource
+
+#pragma mark Initializing a Source
+
+/**
+ Returns a georeferenced image source with an identifier and coordinates
```

Comment:
The more I think about this, it doesn't make sense to partially initialize this type of source. Will just remove this initializer.